src/java.desktop/share/classes/javax/swing/plaf/basic/BasicDirectoryModel.java

```diff
@@ -536,31 +528,30 @@ private final class DoChangeContents implements Runnable {
             this.fid = fid;
         }
 
-        synchronized void cancel() {
-            doFire = false;
-        }
+        @Override
+        public void run() {
+            if (fetchID.get() != fid) {
+                return;
+            }
 
-        public synchronized void run() {
-            if (fetchID.get() == fid && doFire) {
-                int remSize = (remFiles == null) ? 0 : remFiles.size();
-                int addSize = (addFiles == null) ? 0 : addFiles.size();
-                synchronized(fileCache) {
-                    if (remSize > 0) {
-                        fileCache.removeAll(remFiles);
-                    }
-                    if (addSize > 0) {
-                        fileCache.addAll(addStart, addFiles);
-                    }
-                    files = null;
-                    directories = null;
+            int remSize = (remFiles == null) ? 0 : remFiles.size();
+            int addSize = (addFiles == null) ? 0 : addFiles.size();
+            synchronized(fileCache) {
```

Comment:
```suggestion
            synchronized (fileCache) {
```

Reply:
I didn't modify this code. I went ahead and updated it so that it's consistent in the class.